File: Source Code/FlowerShop/src/app/service/bouquet.service.ts
import { Injectable } from "@angular/core";
import { BaseURLService } from "./baseurl.service";
import { HttpClient } from "@angular/common/http";
import { lastValueFrom } from "rxjs";
import { Bouquet } from "../models/bouquet.model";

@Injectable()
export class bouquetService{
 constructor(
  private baseURLService: BaseURLService,
  private httpClient: HttpClient
 ){}

 async findNewBouquet(){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findnewbouquet'));
}

async findFavoriteBouquet(){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findfavoritebouquet'));
}

async findCategoryIdBouquet(){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findcategoryidbouquet'));
}

async findAll(){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findall'));
}

async findByCategoryId(categoryId: string){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findByCategoryId/'+categoryId));
}

async findByPrice(price: string){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findByPriceRange/'+price));
}

async findByEventId(eventId: string){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findByEventId/'+eventId));
}

async findByFavorite(){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findfavoritebouquet'));
}

async findByBouquetId(bouquetId: string){
  return await lastValueFrom(this.httpClient.get(this.baseURLService.getBaseUrl()+'bouquet/findByBouquetId/'+bouquetId));
}
}

File: Source Code/DemoOnlineFloralDelivery/DemoOnlineFloralDelivery/Service/BouquetService.cs
using DemoOnlineFloralDelivery.Models;

namespace DemoOnlineFloralDelivery.Service;

public interface BouquetService
{
    //method hiện thị danh sách bó hoa(Bouquet) thoe ngày tạo mới nhất
    public dynamic findAll();

    // method hiện thị danh sách bó hoa thoe status là true hoặc false
    public dynamic findByStatus(bool status);
    // method hiện thị danh sách tìm kiếm thoe giá của sản phẩm
    public dynamic findByPriceRange(int maxPrice);
    // method tìm kiếm bouquet theo categoryId
    public dynamic findByCategoryId(int categoryId);

    // method tìm kiếm bouquet theo categoryId
    public dynamic findByBouquetName(string bouquetName);
    // created bouquet
    public bool Create(Bouquet bouquet);
    public bool Update(Bouquet bouquet);

    // tìm kiếm bouquet
    public dynamic findById(int bouquetId);

    // method ở giao diện user
    public dynamic findNewBouquet();

    public dynamic findFavoriteBouquet();

    public dynamic findCateogryBouquet();

    public dynamic findByEventId(int eventId);

    public dynamic findByBouquetId(int bouquetId);

}
